torch/_dynamo/variables/misc.py

```diff
@@ -374,25 +373,27 @@ def create(target_values, kwargs):
         kwargs.clear()
 
         for key in ["device_type", "dtype", "enabled", "cache_enabled"]:
-            if isinstance(bound_args.arguments[key], VariableTracker):
-                target_values.append(bound_args.arguments[key])
+            arg = bound_args.arguments[key]
+            if isinstance(arg, VariableTracker):
+                if not isinstance(arg, ConstantVariable):
+                    raise unimplemented("autocast with non-constant arguments")
+                target_values.append(bound_args.arguments[key].as_python_constant())
             else:
-                target_values.append(
-                    variables.ConstantVariable(bound_args.arguments[key])
-                )
+                target_values.append(bound_args.arguments[key])
 
         var = AutocastModeVariable(target_values, initial_values=None, **kwargs)
         return var
 
     def __init__(self, target_values, initial_values=None, **kwargs):
+        mode = kwargs.pop("mode", None)
         super().__init__(
             target_values=target_values, initial_values=initial_values, **kwargs
         )
-        self.target_values = [val.as_python_constant() for val in target_values]
-        self.mode = None
+        self.target_values = target_values
+        self.mode = mode
 
     def exit(self, tx, *args):
-        tx.output.create_node(
+        self.mode = tx.output.create_node(
```

Comment:
Why do we need this change? ```exit_functional_autocast``` doesn't return ```mode```.

Reply:
Well, perhaps, but I want to reset mode to `None` at least.